integreat_cms/cms/utils/totp_utils.py

```diff
@@ -0,0 +1,59 @@
+"""
+This module contains helper functions that are used within the TOTP registration and authentication.
+"""
+
+import base64
+import io
+import pyotp
+import qrcode
+
+from django.conf import settings
+
+
+def generate_totp_qrcode(key, user):
+    """
+    Render event form for HTTP GET requests
+
+    :param key: The required key to generate the QR code
+    :type key: str
+
+    :param user: The user account that is trying to add TOTP to its account
+    :type user: ~django.contrib.auth.models.User
+
+    :return: The QR code as a base64-encoded string
+    :rtype: str
+    """
+    domain = settings.BRANDING.capitalize()
+    email = user.email
+
+    qr_string = f"otpauth://totp/{domain}:{email}?secret={key}&issuer={domain}&algorithm=SHA1&digits=6&period=30"
+
+    img = qrcode.make(qr_string)
+
+    img_byte_arr = io.BytesIO()
+    img.save(img_byte_arr, format="PNG")
+    img_byte_arr = img_byte_arr.getvalue()
+
+    img_base64_string = "data:image/png;base64," + base64.b64encode(
+        img_byte_arr
+    ).decode("utf-8")
+
+    return img_base64_string
+
+
+def check_totp_code(given_totp_code, encoded_key):
+    r"""
+    Helper method to check the given code with the
+
+    :param given_totp_code: The entered TOTP code form the user
+    :type given_totp_code: str
+
+    :param encoded_key: The key that is used for the generation of the true code
+    :type encoded_key: str
+
+    :return: True if the given code matches the initiated
+    :rtype: bool
+    """
```

Comment:
```suggestion
    """
    if not key:
        return False
```
I know, this is probably overly cautious, but since an empty string is a valid totp seed according to the spec, I would just avoid these complications altogether...